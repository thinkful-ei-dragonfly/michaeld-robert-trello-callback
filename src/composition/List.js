import React from 'react';
import Card from './Card';

function List(props) {
  const cards = props.cards
  const cardLists = cards.map(card =>
    
    <Card key={card.id} title={card.title} content={card.content}/>)
  return (
    <section className='List'>
      <header class="List-header">
        <h2>{props.header}</h2>
      </header>
      <div className='List-cards'>
        {cardLists}
        <button type="button" class="List-add-button">
            + Add Random Card
          </button>
      </div>
    </section>

  )
}

export default List;