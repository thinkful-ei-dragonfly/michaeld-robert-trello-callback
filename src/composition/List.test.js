import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import '../index.css';
import STORE from '../store';
import List from './List';

describe('<List />', () => {
  it('renders without crashing', () => {
      // Create a DOM element to render the component into
      const div = document.createElement('div');
      // Render the component
      // If something is wrong it will fail here
      ReactDOM.render(<List key={STORE.lists.id} header={STORE.lists.header} cards={STORE.lists}/>, div);
      // Clean up
    ReactDOM.unmountComponentAtNode(div);
  })
  it('renders without crashing', () => {
    // Render the component, as JSON
    const tree = renderer.create(<List key={STORE.lists.id} header={STORE.lists.header} cards={STORE.lists}/>).toJSON();
    // Check whether it matches the previous snapshot
    // Stored in __snapshots__/App.test.js.snap
    expect(tree).toMatchSnapshot();
  });
});