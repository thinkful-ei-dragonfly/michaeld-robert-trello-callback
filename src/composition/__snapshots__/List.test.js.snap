// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders without crashing 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2 />
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="delete"
      >
        delete
      </button>
      <h3 />
      <p />
    </div>
    <div
      className="Card"
    >
      <button
        type="delete"
      >
        delete
      </button>
      <h3 />
      <p />
    </div>
    <div
      className="Card"
    >
      <button
        type="delete"
      >
        delete
      </button>
      <h3 />
      <p />
    </div>
    <div
      className="Card"
    >
      <button
        type="delete"
      >
        delete
      </button>
      <h3 />
      <p />
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
